q2_moshpit/eggnog/tests/test_method.py

```diff
@@ -72,6 +77,31 @@ def test_good_small_search_mags(self):
 
         pdt.assert_frame_equal(obs, exp)
 
+    def test_good_small_search_mags(self):
+        mags = qiime2.Artifact.import_data(
+            'SampleData[MAGs]',
+            self.get_data_path('mag-sequences-per-sample')
+        ).view(MultiMAGSequencesDirFmt)
+
+        _, obs = _eggnog_diamond_search(
+            sequences=mags,
+            diamond_db=self.diamond_db
+        )
+        exp = pd.DataFrame(
+            {
+                '8': [3.0, 3.0, 0.0, 0.0],
+                '0': [0.0, 0.0, 1.0, 1.0],
+                '2': [0.0, 0.0, 1.0, 1.0]
+            },
+            index=['194b1aca-9373-4298-ba5c-05b382d1f553',
+                   '8f40a3bc-14f0-426b-b2ba-7fb3b1cd0c02',
+                   'c6bd5123-35cf-473a-bebf-85cf544bcd48',
+                   'e1ba1d20-b466-4fef-ae19-6bf9c5c63d6f']
+        )
+        exp.columns.name = 'sseqid'
+
+        pdt.assert_frame_equal(obs, exp)
+
```

Comment:
Is there a reason why here and in the other tests we actually run the search (i.e. call `_eggnog_diamond_search` without patching anything)? Would it not be preferable to patch the` subprocess.run` calls?

Reply:
Hey @Sann5, yeah... This is intentional - we had discussed a while ago that for Kraken and Eggnog we'll include some tests that actually run the entire action on a minimal dataset to be able to detect potential regressions in case anything changes within those tools.